node/src/exchange/utils.rs

```diff
@@ -58,11 +64,63 @@ impl HeaderResponseExt for HeaderResponse {
     }
 }
 
-pub(super) trait ExtendedHeaderExt {
-    fn to_header_response(&self) -> HeaderResponse;
+pub trait ExtendedHeaderExt {
+    fn with_height(height: u64) -> ExtendedHeader;
```

Comment:
I've canibalised this trait and moved conversion to header response to `ToHeaderResponse` as I want to implement that for easier Result transformation.
This Header generation is used currently only in tests, not sure what's a good place for it.

Reply:
I believe this shouldn't be here because it generates an invalid `ExtendedHeader` which its current purpose is ti be used in tests. I'm in favor of something like:

```rust
fn gen_extended_header(height: u64) -> ExtendedHeader
```

Also I prefer `to_header_response` in `ExtendedHeaderExt`. I can not see why it had to be changed.

Reply:
My idea was to have `ToHeaderResponse` solely for converting stuff to HeaderResponse, as I want to use that to [convert Result from store into respose ](https://github.com/eigerco/celestia-node-rs/pull/72/files#diff-93f633bcbe6080e868774f979d9cae6ffa477bcc909c263b66d42372172176d6R138-R153). 
I see your point this not generating valid headers, so moved this to test as a freestanding function.

Reply:
I don't like the convertion of Result into response. I would prefer an explicit function if it going to be used in many places. If it's just one place then I prefer `match`.  If your intention was to have a cleaner logic before sending the result, one idea is to add this constructor in `HeaderResponseExt`:

```rust
fn not_found() -> HeaderResponse;
```

Also, we should never send Invalid. That was added intetionally because of Protobuf convertion rules.

Reply:
I think Invalid actually can be send back if the Request is invalid. We can try to send an empty HeaderRequest to the network and see if it behaves that way.

Reply:
Yep! I'm sending `Invalid`, if there's something wrong with the request and `NotFound`, if, well, not found or if there's an issue retrieving the value from store (I think it makes more sense than returning Invalid, if request is correct but there might be issues on our side).

This seems consistent with go node, I see `Invalid` back for empty or nonsensical requests.